packages/main/src/sync.ts

```diff
@@ -338,25 +338,18 @@ class BoxSync {
 }
 
 function matchesCachePolicy(message: imap.ImapMessageAttributes): boolean {
-  let labelMatches = true
-  let flagMatches = true
-  let dateMatches = true
+  const labels = message["x-gm-labels"] || []
+  const labelMatches =
+    cachePolicy.labels &&
+    !cachePolicy.labels.some(label => labels.includes(label))
```

Comment:
Yes, I like this style better.

But now I'm noticing that the logic is inverted. `labelMatches` should be `true` if `cachePolicy.labels` is truthy, and some label in the cache policy is also in `labels`. I think the `!` should be removed from this expression, and from the check for flags.